File: backend/LeetWars.Core/LeetWars.Core.DAL/Entities/ChallengeVersion.cs
using LeetWars.Core.DAL.Enums;

namespace LeetWars.Core.DAL.Entities
{
    public class ChallengeVersion : AuditEntity<long>
    {
        public long LanguageId { get; set; }
        public long ChallengeId { get; set; }
        public string InitialSolution { get; set; }
        public string CompleteSolution { get; set; }
        public string PreloadedCode { get; set; }
        public string TestCases { get; set; }
        public string ExampleTestCases { get; set; }

        public ChallengeStatus Status { get; set; }
        public ICollection<UserSolution> Solutions { get; } = new List<UserSolution>();
        public Language? Language { get; set; }
        public User? Author { get; set; }


        public ChallengeVersion(string initialSolution, string completeSolution, string preloadedCode, string testCases,
            string exampleTestCases)
        {
            InitialSolution = initialSolution;
            CompleteSolution = completeSolution;
            PreloadedCode = preloadedCode;
            TestCases = testCases;
            ExampleTestCases = exampleTestCases;
        }
    }
}

File: frontend/src/app/shared/models/challenge-version/challenge-version.ts
import { ChallengeStatus } from '@shared/enums/challenge-status';

import { AuditEntity } from '../common/audit-entity/audit-entity';
import { ILanguageFull } from '../language/languageFull';
import { ILanguageVersion } from '../language-version/language-version';
import { IUserSolution } from '../profile/user-solution';
import { ITest } from '../test/test';
import { IUser } from '../user/user';

export interface IChallengeVersion extends AuditEntity<number> {
    languageId: number;
    challengeId: number;
    initialSolution: string;
    completeSolution: string;
    preloadedCode: string;
    testCases: string;
    exampleTestCases: string;
    status: ChallengeStatus;
    languageVersions: Array<ILanguageVersion>;
    solutions: Array<IUserSolution>;
    tests: Array<ITest>;
    language: ILanguageFull;
    author: IUser;
}
